Assignment.Tests/PingProcess.Tests.cs

```diff
@@ -0,0 +1,232 @@
+using IntelliTect.TestTools;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using System.Text;
+using Assignment;
+
+
+namespace Assignment.Tests;
+
+[TestClass]
+public class PingProcessTests
+{
+    PingProcess Sut { get; set; } = new();
+
+    [TestInitialize]
+    public void TestInitialize()
+    {
+        Sut = new();
+    }
+
+    [TestMethod]
+    public void Start_PingProcess_Success()
+    {
+        Process process = Process.Start("ping", "localhost");
+        process.WaitForExit();
+        Assert.AreEqual<int>(0, process.ExitCode);
+    }
+
+    [TestMethod]
+    public void Run_GoogleDotCom_Success()
+    {
+        PingResult result = Sut.Run("google.com");
+        Assert.AreEqual<int>(0, result.ExitCode);
+    }
+
+
+    [TestMethod]
+    public void Run_InvalidAddressOutput_Success()
+    {
+        (int exitCode, string? stdOutput) = Sut.Run("badaddress");
+        Assert.IsFalse(string.IsNullOrWhiteSpace(stdOutput));
+        stdOutput = WildcardPattern.NormalizeLineEndings(stdOutput!.Trim());
+        Assert.AreEqual<string?>(
+            "Ping request could not find host badaddress. Please check the name and try again.".Trim(),
+            stdOutput,
+            $"Output is unexpected: {stdOutput}");
+        Assert.AreEqual<int>(1, exitCode);
+    }
+
+    [TestMethod]
+    public void Run_CaptureStdOutput_Success()
+    {
+        PingResult result = Sut.Run("localhost");
+        AssertValidPingOutput(result);
+    }
+
+    [TestMethod]
+    public void RunTaskAsync_Success()
+    {
+        PingProcess pingProcess = new();
+        string hostNameOrAddress = "localhost";
+
+
+        var pingTask = PingProcess.RunTaskAsync(hostNameOrAddress);
+
+        var result = pingTask.Result;
+        Assert.IsNotNull(result.StdOutput);
+        Assert.IsTrue(result.ExitCode == 0);
+    }
+    [TestMethod]
+    public void RunAsync_UsingTaskReturn_Success()
+    {
+
+        PingProcess pingProcess = new();
+        string hostNameOrAddress = "localhost";
+
+
+        var task = PingProcess.RunTaskAsync(hostNameOrAddress);
+        var res = task.Result;
+
+
+        Assert.IsNotNull(res.StdOutput);
+        Assert.IsTrue(res.ExitCode == 0);
```

Comment:
Can simplify, no reason to use a ```var``` since your method should be returning a ```Task<PingResult>```. You could even remove the creation of ```res``` and do the following
```suggestion
        AssertValidPingOutput(task.Result);
```